Reconcile BTP Manager Secret (#2675)
kyma-project/control-plane: docs/kyma-environment-broker/03-17-runtime-reconciler.md

# Runtime reconciler

Runtime reconciler is application which perform some reconcile tasks on runtimes (SKRs).

## Overview

> Currently there is one task for runtime reconciler, it is to reconcile BTP Manager secrets on SKRs. It is achived in two ways, first one is implemention with usage of runtime-watcher, which in real time sends event about changes of secret from SKR to KEB. The second method is a job, which periodically loop over all instances from KEB database which have existing assigned runtimes ID, and for each of them, do a check if secret on SKR match with credentials from KEB database.

Reviewer: ```suggestion
Currently, there is one task for Runtime Reconciler. It reconciles BTP Manager Secrets on SAP BTP, Kyma runtimes (SKRs). It can do it in two ways: 
- by implementation with the usage of Runtime Watcher, which sends events about changes of the Secret from the SKR to KEB in real time. 
- with a job, which periodically loops over all instances from the KEB database; each instance has an existing assigned runtime ID; the job checks if the Secret on the SKR matches the credentials from the KEB database.
```

## Configuration

The application is defined as kubernetes deployment.

Use the following environment variables to configure the application:

| Environment variable                                             | Description                                                                                                                      | Default value |
| ---------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ------------- |

> | **RUNTIME_RECONCILER_WATCHER_ENABLED**                           | Specifies whether application should use watcher to reconcile.                                                                   | `false`        |

Reviewer: ```suggestion
| **RUNTIME_RECONCILER_WATCHER_ENABLED**                           | Specifies whether the application should use Runtime Watcher for reconciliation.                                                                   | `false`        |
```

| **RUNTIME_RECONCILER_JOB_ENABLED**                               | Specifies whether application should use job to reconcile.                                                                       | `false`        |
| **RUNTIME_RECONCILER_DRY_RUN**                                   | Specifies whether to run the application in the dry-run mode.                                                                    | `true`        |

> | **RUNTIME_RECONCILER_BTP_MANAGER_SECRET_WATCHER_ADDR**           | Specifies port of watcher.                                                                                                       | `0`           |

Reviewer: ```suggestion
| **RUNTIME_RECONCILER_BTP_MANAGER_SECRET_WATCHER_ADDR**           | Specifies Runtime Watcher's port.                                                                                                       | `0`           |
```

> | **RUNTIME_RECONCILER_BTP_MANAGER_SECRET_WATCHER_COMPONENT_NAME** | Specifies company name of watcher.                                                                                               | `NA`          |

Reviewer: ```suggestion
| **RUNTIME_RECONCILER_BTP_MANAGER_SECRET_WATCHER_COMPONENT_NAME** | Specifies the component name for Runtime Watcher.                                                                                               | `NA`          |
```

> | **RUNTIME_RECONCILER_AUTO_RECONCILE_INTERVAL**                   | Specifies in what interval (in hours), the job should run.                                                                       | `24`          |

Reviewer: ```suggestion
| **RUNTIME_RECONCILER_AUTO_RECONCILE_INTERVAL**                   | Specifies at what intervals the job runs  (in hours).                                                                       | `24`          |
```

# Runtime Reconciler

Runtime Reconciler is an application that performs reconciliation tasks on SAP BTP, Kyma runtimes (SKRs).

## Overview

> Currently, there is one task for Runtime Reconciler. It reconciles BTP Manager Secrets on Kyma runtimes. It can do it in two ways: 

Reviewer: ```suggestion
Currently, there is one task for Runtime Reconciler. It reconciles BTP Manager Secrets on Kyma Runtimes. It can do it in two ways: 
```

> - by implementation with the usage of Runtime Watcher, which sends events about changes of the Secret from Kyma runtime to KEB in real time. 

Reviewer: ```suggestion
- by implementation with the usage of Runtime Watcher, which sends events about changes of the Secret from Kyma Runtime to KEB in real time. 
```

> - with a job, which periodically loops over all instances from the KEB database; each instance has an existing assigned runtime ID; the job checks if the Secret on the SKR matches the credentials from the KEB database.

Reviewer: ```suggestion
- with a job, which periodically loops over all instances from the KEB database; each instance has an existing assigned Runtime ID; the job checks if the Secret on the Kyma Runtime matches the credentials from the KEB database.
```
